### PR TITLE
Greenlight v3 recording ready url

### PR DESCRIPTION
**Pull request by PhilippKilian**
* correct implementation of recording ready callback
* update docker base images

### CODE REVIEW

File: b3lb/rest/task/recording.py [hunk 1]
```diff
@@ -79,18 +81,21 @@ def render_record(record_set: RecordSet):
...
             else:
                 for record_profile in RecordProfile.objects.filter(is_default=True):
-                    render_by_profile(record_set, record_profile, tempdir)
+                    record_id = render_by_profile(record_set, record_profile, tempdir)
+
+    # Greenlight v3 sends a record ready URL, Greenlight v2 doesn't send one
+    if record_set.recording_ready_origin_url:
+        body = {"signed_parameters": jwt_encode({"meeting_id": record_set.meta_meeting_id, "record_id": record_id}, record_set.secret.get_valid_secret)}
+        response = post(record_set.recording_ready_origin_url, json=body)
+        print(f"Send record ready callback to: {record_set.recording_ready_origin_url} with return code: {response.status_code}")
```
Comment (liske) [inline b3lb/rest/task/recording.py:93]: Looks like orphan debugging print?

File: b3lb/rest/task/recording.py [hunk 2]
```diff
@@ -79,18 +81,21 @@ def render_record(record_set: RecordSet):
...
+        if response.status_code < 200 or 299 < response.status_code:
```
Comment (liske) [inline b3lb/rest/task/recording.py:94]: What about `300` codes? Wouldn't `response.ok` be sufficient?

File: b3lb/rest/models.py [hunk 3]
```diff
@@ -400,6 +400,12 @@ class Meta(object):
     def __str__(self):
         return "{}-{}".format(self.tenant.slug, str(self.sub_id).zfill(3))
 
+    @property
+    def get_valid_secret(self) -> str:
```
Comment (liske) [inline b3lb/rest/models.py:404]: Prefixing the name of a property with `get_` irritates me.

File: b3lb/rest/models.py [hunk 4]
```diff
@@ -400,6 +400,12 @@ class Meta(object):
...
+        if self.secret2:
```
Comment (liske) [inline b3lb/rest/models.py:405]: This is fragile and may break key rollover. It requires to use the secret that was used to create the meeting, didn't it?

**Review by liske — CHANGES_REQUESTED** [review]

File: b3lb/rest/task/recording.py [hunk 5]
```diff
@@ -79,18 +81,21 @@ def render_record(record_set: RecordSet):
...
             if profile_relations.count() > 0:
                 for profile_relation in profile_relations:
-                    render_by_profile(record_set, profile_relation.record_profile, tempdir)
+                    record_id = render_by_profile(record_set, profile_relation.record_profile, tempdir)
             else:
                 for record_profile in RecordProfile.objects.filter(is_default=True):
-                    render_by_profile(record_set, record_profile, tempdir)
+                    record_id = render_by_profile(record_set, record_profile, tempdir)
+
+    # Greenlight v3 sends a record ready URL, Greenlight v2 doesn't send one
```
[the remark this answers is not in the document]
Comment (PhilippKilian) [reply b3lb/rest/task/recording.py:89]: It't generic... I will update the comment

File: b3lb/rest/task/recording.py (hunk 1 above)
Comment (PhilippKilian) [reply b3lb/rest/task/recording.py:93]: Looks like but it's just a logging print

File: b3lb/rest/task/recording.py (hunk 2 above)
Comment (PhilippKilian) [reply b3lb/rest/task/recording.py:94]: No, 300 codes are treated as errors:

https://docs.bigbluebutton.org/development/api/#recording-ready-callback-url

File: b3lb/rest/models.py (hunk 3 above)
Comment (PhilippKilian) [reply b3lb/rest/models.py]: obsolete, see secret fragility

**Pull request merged**